clipdel

```diff
@@ -16,72 +16,164 @@ cache_file_prefix=$cache_dir/line_cache
 lock_file=$cache_dir/lock
 lock_timeout=2
 
-if [[ $1 == --help ]] || [[ $1 == -h ]]; then
-    cat << 'EOF'
-clipdel deletes clipmenu entries matching a regex. By default, just lists what
-it would delete, pass -d to do it for real.
+line_cache_files=( "$cache_file_prefix"_* )
+
+check_conflict() {
+    if [[ -z "$CM_FUNCTION" ]]; then
+        CM_FUNCTION="$1"
+        CM_REQUIRE_PATTERN=$2
+        if ! ((CM_REQUIRE_PATTERN)) && ((${#CM_PATTERNS[@]})); then
+            printf '%s\n' "Too many arguments" >&2
+            return 2
+        fi
+    else
+       printf '%s\n' "${3:-Options '$1' and '$CM_FUNCTION' conflict}" >&2
+       return 1
+    fi
+}
 
-".*" is special, it will just nuke the entire data directory, including the
-line caches and all other state.
+print_usage() {
+    cat << 'EOF'
+Usage: clipdel [OPTION] [REGEX]
+Delete clipmenu entries.
 
 Arguments:
 
-    -d  Delete for real.
+    --all                      Delete all entries
+    -c, --current              Delete current entry
+    -d, --delete               Delete for real (default when deleting all, current or from stdin)
```

Comment:
Hmm, not sure the default value of `-d` should vary depending on the other options. Can we still keep the sanity check, please? :-)

I agree that this probably is not super useful in these cases, but it seems more confusing to vary it.